CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -354,6 +354,64 @@ local function CreateMenuBar3D(barAlignment, threeDPanel)
 	return this
 end
 
+local function CreateMenuChangedNotifier()
+	local this = {}
+	local notifier3D = require(GuiRoot.Modules.VR.NotifierHint3D)
+
+	function this:PromptNotification()
+		-- Don't show the notification if we are looking down at the menubar already
+		if not (TopbarPanel3D:IsVisible() and TopbarPanel3D.isEnabled) then
+			notifier3D:BeginNotification(notifier3D.DEFAULT_DURATION)
+		end
+	end
+
+	spawn(function()
+		-- Spawn this because we don't want circular requires
+		local backpack3D = require(GuiRoot.Modules.BackpackScript3D)
+		backpack3D.ToolAddedEvent.Event:connect(function() this:PromptNotification() end)
+	end)
+
+	PlayersService.FriendRequestEvent:connect(function(fromPlayer, toPlayer, eventStatus)
+		if toPlayer == PlayersService.LocalPlayer then
+			this:PromptNotification()
+		end
+	end)
+
+	local function findScreenGuiAncestor(object)
+		if not object then
+			return nil
+		end
+		local parent = object.Parent
+		if parent and parent:IsA('ScreenGui') then
+			return parent
+		end
+		return hasScreenGuiAncestor(parent)
```

Comment:
Wasn't able to find the declaration of this function, was this meant to be recursive and be findScreenGuiAncestor?